v2/api/network/customizations/load_balancer_extension.go

```diff
@@ -80,94 +80,3 @@ func getInboundNatRuleGVK(lb genruntime.ARMMetaObject) schema.GroupVersionKind {
 
 	return gvk
 }
-
-// fuzzySetSubnets assigns a collection of subnets to the subnets property of the loadBalancer. Since there are
-// many possible ARM API versions and we don't know which one we're using, we cannot do this statically.
-// To make matters even more horrible, the type used in the loadBalancer.properties.inboundNatRule property is not the same
-// type as used for inboundNatRule.properties (although structurally they are basically the same). To overcome this
-// we JSON serialize the subnet and deserialize it into the loadBalancer.properties.inboundNatRule field.
-func fuzzySetInboundNatRules(lb genruntime.ARMResourceSpec, inboundNatRules []genruntime.ARMResourceSpec) (err error) {
-	if len(inboundNatRules) == 0 {
-		// Nothing to do
-		return nil
-	}
-
-	defer func() {
-		if x := recover(); x != nil {
-			err = errors.Errorf("caught panic: %s", x)
-		}
-	}()
-
-	// Here be dragons
-	lbValue := reflect.ValueOf(lb)
-	lbValue = reflect.Indirect(lbValue)
-	if (lbValue == reflect.Value{}) {
-		return errors.Errorf("cannot assign to nil loadbalancer")
-	}
-
-	propertiesField := lbValue.FieldByName("Properties")
-	if (propertiesField == reflect.Value{}) {
-		return errors.Errorf("couldn't find properties field on loadbalancer")
-	}
-
-	propertiesValue := reflect.Indirect(propertiesField)
-	if (propertiesValue == reflect.Value{}) {
-		// If the properties field is nil, we must construct an entirely new properties and assign it here
-		temp := reflect.New(propertiesField.Type().Elem())
-		propertiesField.Set(temp)
-		propertiesValue = reflect.Indirect(temp)
-	}
-
-	inboundNatRulesField := propertiesValue.FieldByName("InboundNatRules")
-	if (inboundNatRulesField == reflect.Value{}) {
-		return errors.Errorf("couldn't find InboundNatRules field on LoadBalancer")
-	}
-
-	if inboundNatRulesField.Type().Kind() != reflect.Slice {
-		return errors.Errorf("inboundNatRules field on LoadBalancer was not of kind Slice")
-	}
-
-	elemType := inboundNatRulesField.Type().Elem()
-	inboundNatRuleSlice := reflect.MakeSlice(inboundNatRulesField.Type(), 0, 0)
-
-	for _, inboundNatRule := range inboundNatRules {
-		embeddedInboundNatRule := reflect.New(elemType)
-		err := fuzzySetInboundNatRule(inboundNatRule, embeddedInboundNatRule)
-		if err != nil {
-			return err
-		}
-
-		inboundNatRuleSlice = reflect.Append(inboundNatRuleSlice, reflect.Indirect(embeddedInboundNatRule))
-	}
-
-	// Now do the assignment. We do it differently here, as we need to make sure to retain current/updated/deleted InboundNatRules present on the LoadBalancer.
```

Comment:
Was this preserved?

Reply:
Yes it is [here](https://github.com/Azure/azure-service-operator/pull/3121/files/8d3e16092fa5137c42f4f6609df3018a5c56ee7a#diff-d201efade7573c78d819afb69f08b9166be871ed72e998625e68bd89bc163f29R184)

Reply:
Yes it is [here](https://github.com/Azure/azure-service-operator/pull/3121/files#diff-d201efade7573c78d819afb69f08b9166be871ed72e998625e68bd89bc163f29R184)